code/api/src/main/scala/com/avast/datadog4s/api/MetricFactory.scala

```diff
@@ -1,15 +1,30 @@
 package com.avast.datadog4s.api
 
+import com.avast.datadog4s.api.MetricFactory.TimerMode.HistogramTimer
+import com.avast.datadog4s.api.MetricFactory.TimerMode
 import com.avast.datadog4s.api.metric.{ Count, Timer, UniqueSet }
 
 trait MetricFactory[F[_]] {
   def histogram: HistogramFactory[F]
   def distribution: DistributionFactory[F]
   def gauge: GaugeFactory[F]
-  def timer(aspect: String, sampleRate: Option[Double] = None): Timer[F]
+  def timer(
+    aspect: String,
+    sampleRate: Option[Double] = None,
+    timerMode: TimerMode = HistogramTimer: TimerMode
+  ): Timer[F]
   def count(aspect: String, sampleRate: Option[Double] = None): Count[F]
   def uniqueSet(aspect: String): UniqueSet[F]
 
   def withTags(tags: Tag*): MetricFactory[F]
   def withScope(name: String): MetricFactory[F]
+  def withTimerMode(timerMode: TimerMode): MetricFactory[F]
+}
+
+object MetricFactory {
```

Comment:
👍 

I was also considering an approach where this choice would be reified in the factory type - i.e. a new class that uses Distribution, but I think the config-driven change you've made here is actually truer to the idioms of this library.

Reply:
yeah, somehow i've learn to value simplicity over fanciness over last few years when working with scala :)

Reply:
> This would definitely work for us

awesome, let me polish it a bit, add docs etc and hopefully it will be ready on monday, also waiting for a review from @hanny24